### PR TITLE
[1.18] Cherry pick of #98477: update gogo/protobuf to v1.3.2

### PR DESCRIPTION
**Pull request by joelsmith**
Cherry pick of #98477 on release-1.18.

#98477: update gogo/protobuf to v1.3.2

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note
Regenerate protobuf code to fix CVE-2021-3121
```

### CODE REVIEW

**Comment by sttts** [comment]
automated cherry-pick with generated code are suspicious. Have you checked that updating the proto files afterwards is a noop?

**Comment by sttts** [comment]
/lgtm
/approve

for dep changes.

/assign @liggitt 

for api changes.

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/meta.go [hunk 1]
```diff
@@ -156,7 +156,9 @@ func (meta *ObjectMeta) GetDeletionTimestamp() *Time { return meta.DeletionTimes
 func (meta *ObjectMeta) SetDeletionTimestamp(deletionTimestamp *Time) {
 	meta.DeletionTimestamp = deletionTimestamp
 }
-func (meta *ObjectMeta) GetDeletionGracePeriodSeconds() *int64 { return meta.DeletionGracePeriodSeconds }
+func (meta *ObjectMeta) GetDeletionGracePeriodSeconds() *int64 {
+	return meta.DeletionGracePeriodSeconds
+}
```
Comment (liggitt) [inline staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/meta.go:161]: revert formatting changes

File: staging/src/k8s.io/apimachinery/pkg/fields/selector.go [hunk 2]
```diff
@@ -57,13 +57,15 @@ type Selector interface {
 
 type nothingSelector struct{}
 
-func (n nothingSelector) Matches(_ Fields) bool                                      { return false }
```
Comment (liggitt) [inline staging/src/k8s.io/apimachinery/pkg/fields/selector.go:60]: revert formatting changes

File: staging/src/k8s.io/apimachinery/pkg/labels/selector.go [hunk 3]
```diff
@@ -68,13 +68,15 @@ func Everything() Selector {
 
 type nothingSelector struct{}
 
-func (n nothingSelector) Matches(_ Labels) bool                                      { return false }
```
Comment (liggitt) [inline staging/src/k8s.io/apimachinery/pkg/labels/selector.go:71]: revert formatting changes

File: staging/src/k8s.io/code-generator/cmd/informer-gen/generators/generic.go [hunk 4]
```diff
@@ -73,9 +73,11 @@ type group struct {
 
 type groupSort []group
 
-func (g groupSort) Len() int           { return len(g) }
```
Comment (liggitt) [inline staging/src/k8s.io/code-generator/cmd/informer-gen/generators/generic.go:76]: revert formatting changes

File: staging/src/k8s.io/kubectl/pkg/util/templates/markdown.go [hunk 5]
```diff
@@ -111,22 +111,24 @@ func (r *ASCIIRenderer) TableRow(out *bytes.Buffer, text []byte)
 func (r *ASCIIRenderer) TableHeaderCell(out *bytes.Buffer, text []byte, align int)        { r.fw(out, text) }
 func (r *ASCIIRenderer) TableCell(out *bytes.Buffer, text []byte, align int)              { r.fw(out, text) }
 func (r *ASCIIRenderer) Footnotes(out *bytes.Buffer, text func() bool)                    { text() }
-func (r *ASCIIRenderer) FootnoteItem(out *bytes.Buffer, name, text []byte, flags int)     { r.fw(out, text) }
```
Comment (liggitt) [inline staging/src/k8s.io/kubectl/pkg/util/templates/markdown.go:114]: revert formatting changes

**Comment by liggitt** [comment]
lgtm after formatting changes are reverted

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**